<a href="https://colab.research.google.com/github/Aderson79/dataset_colab/blob/main/Titanic_dataset.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**DATASET TITANIC (KAGGLE)**

##1) Escolha do dataset:

- **Por que escolheram esse dataset?**

   Ele geralmente contém informações sobre os passageiros a bordo do navio Titanic, incluindo se eles sobreviveram ou não ao naufrágio. Cada linha representa um passageiro, e as colunas fornecem detalhes sobre eles

- **Que tipo de análise poderia ser feita com ele?**

  O objetivo comum ao analisar este dataset é prever a sobrevivência de um passageiro com base nas outras características.

- **Quais desafios vocês imaginam encontrar na limpeza desses dados?**

  Valores ausentes, conversão de tipos de dados, variáveis categóricas e colunas irrelevantes.



In [ ]:
import pandas as pd
content_file_path = '/content/train.xlsx'

try:
  df = pd.read_excel(content_file_path)
  print("Sucesso: O dataset 'train.xlsx' foi carregado corretamente!")
except FileNotFoundError:
  print(f"Erro: O arquivo não foi encontrado em '{content_file_path}'. Certifique-se de que ele foi salvo neste local.")
except Exception as e:
  print(f"Ocorreu um erro ao ler o arquivo: {e}")

Sucesso: O dataset 'train.xlsx' foi carregado corretamente!


In [ ]:
print("\nExibindo as primeiras linhas do dataset:\n")
df.head()


Exibindo as primeiras linhas do dataset:



,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22,1,0,A/5 21171,7.25,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38,1,0,PC 17599,712833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26,0,0,STON/O2. 3101282,7925,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35,1,0,113803,53.1,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35,0,0,373450,2026-05-08 00:00:00,NaN,S


In [ ]:
print("\nExibindo a dimensão do dataset:\n")
df.shape


Exibindo a dimensão do dataset:



(891, 12)

In [ ]:
print("\nExibindo os tipos de dados do dataset:\n")
df.dtypes


Exibindo os tipos de dados do dataset:



,0
PassengerId,int64
Survived,int64
Pclass,int64
Name,object
Sex,object
Age,object
SibSp,int64
Parch,int64
Ticket,object
Fare,object


## 2) Quantas colunas e quantas linhas existem no dataset?

  O dataset possui 891 linhas e 12 colunas.
## 3) Quais tipos de dados existem em cada coluna?

  - PassengerId: int64;

  - Survived: int64;

  - Pclass: int64;

  - Name: object;

  - Sex: object;

  - Age: object;

  - SibSp: int64;

  - Parch: int64;

  - Ticket: object;

  - Fare: object;

  - Cabin: object;

  - Embarked: object.

## 4) Existem colunas que não fazem sentido para análise? Quais?

  - **PassengerId**: É um identificador único e geralmente não contribui para a predição ou análise de padrões diretamente. Pode ser descartada.

  - **Name**: a string completa não é diretamente utilizável para a maioria das análises.

  - **Ticket**: Assim como o PassengerId, é um identificador único e geralmente não é relevante para a análise direta.

  - **Cabin**: Frequentemente possui muitos valores ausentes e, se não for tratada de forma específica, pode ser difícil de usar.

    **OBS**: Além disso, as colunas Age e Fare estão como tipo object, mas são variáveis numéricas importantes. Elas precisarão ser convertidas para um tipo numérico (float ou int) para que possam ser utilizadas em análises quantitativas.

In [ ]:
print("Verficar valores ausentes:\n")
df.isnull().sum()

Verficar valores ausentes:



,0
PassengerId,0
Survived,0
Pclass,0
Name,0
Sex,0
Age,177
SibSp,0
Parch,0
Ticket,0
Fare,0


In [ ]:
print("Verificar valores duplicados:\n")
df.duplicated().sum()

Verificar valores duplicados:



np.int64(0)

In [ ]:
print("Exibir informações gerais do dataset:\n")
df.info()

Exibir informações gerais do dataset:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   PassengerId  891 non-null    int64 
 1   Survived     891 non-null    int64 
 2   Pclass       891 non-null    int64 
 3   Name         891 non-null    object
 4   Sex          891 non-null    object
 5   Age          714 non-null    object
 6   SibSp        891 non-null    int64 
 7   Parch        891 non-null    int64 
 8   Ticket       891 non-null    object
 9   Fare         891 non-null    object
 10  Cabin        204 non-null    object
 11  Embarked     889 non-null    object
dtypes: int64(5), object(7)
memory usage: 83.7+ KB


##5) Existem valores nulos ? Em quais colunas?

Sim, existem valores nulos nas colunas Age e Embarked, que simbolizam respectivamente o idade e o porto em que o passageiro embarcou.

## 6) O que pode ser feito para tratar esses valores?

Remover as linhas em que Age e Embarked são nulas.

##7) Existem valores duplicados? Como podemos resolver isso?

Não existem valores duplicados no dataset.



##Limpeza dos dados:

In [34]:
print("Remover colunas desnecessárias")
df = df.drop(columns=['PassengerId', 'Name', 'Ticket', 'Cabin'], errors='ignore')

Remover colunas desnecessárias


In [35]:
print("Remover linhas com valores nulos em Age e Embarked")
df = df.dropna(subset=['Age', 'Embarked'])

Remover linhas com valores nulos em Age e Embarked


In [37]:
print("Converter colunas Age e Fare para tipo numérico")
df['Age'] = pd.to_numeric(df['Age'], errors='coerce')
df['Fare'] = pd.to_numeric(df['Fare'], errors='coerce')

Converter colunas Age e Fare para tipo numérico


In [ ]:
print("Exibir informações gerais do dataset após a transfomação do tipo das variáveis:\n")
df.info()

In [40]:
print("Verificando valores nulos apenas nas colunas 'Age' e 'Fare':\n")
df[['Age', 'Fare']].isnull().sum()

Verificando valores nulos apenas nas colunas 'Age' e 'Fare':



,0
Age,8
Fare,208


##Apareceram valores ausentes nas colunas Fare e Age após a transformação para valores numéricos.

**Solução**: Preencher valores nulos nessas colunas com a mediana para permitir a análise sem afetar e sem eliminar as linhas.

In [43]:
print("Preenchendo valores nulos em 'Age' com a mediana...")
df['Age'] = df['Age'].fillna(df['Age'].median())

print("Preenchendo valores nulos em 'Fare' com a mediana...")
df['Fare'] = df['Fare'].fillna(df['Fare'].median())

Preenchendo valores nulos em 'Age' com a mediana...
Preenchendo valores nulos em 'Fare' com a mediana...


##Após a limpeza dos dados:

In [44]:
print("Informações gerais do dataset:\n")
df.info()

Informações gerais do dataset:

<class 'pandas.core.frame.DataFrame'>
Index: 712 entries, 0 to 890
Data columns (total 8 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Survived  712 non-null    int64  
 1   Pclass    712 non-null    int64  
 2   Sex       712 non-null    object 
 3   Age       712 non-null    float64
 4   SibSp     712 non-null    int64  
 5   Parch     712 non-null    int64  
 6   Fare      712 non-null    float64
 7   Embarked  712 non-null    object 
dtypes: float64(2), int64(4), object(2)
memory usage: 50.1+ KB


In [45]:
print("Soma dos valores nulos:\n")
df.isnull().sum()

Soma dos valores nulos:



,0
Survived,0
Pclass,0
Sex,0
Age,0
SibSp,0
Parch,0
Fare,0
Embarked,0


##8) O que foi alterado no dataset?
  - As colunas PassengerId, Name, Ticket e Cabin foram removidas. Elas foram consideradas desnecessárias para a análise direta ou continham muitos valores ausentes (Cabin).

  - Todas as linhas onde Age ou Embarked tinham valores nulos foram descartadas. Isso garante que não haja dados incompletos nessas variáveis cruciais.

  - As colunas Age e Fare foram convertidas para tipos numéricos (float64).

  - Após a conversão (devido a entradas que não puderam ser convertidas para números) foram preenchidos com a mediana de suas respectivas colunas. Isso preserva o maior número possível de dados sem introduzir distorções extremas.

##9) Houve perda significativa de dados ?
Houve uma redução no tamanho total do dataset devido a remoção das linhas onde Age e Embarked eram nulas.

##10) Os dados estão prontos para análise?
Para análise exploratória e descritiva, sim, os dados estão prontos! Não há mais valores nulos, e as variáveis numéricas estão no formato correto. Você já pode começar a visualizar e entender as relações.

##11) Que tipo de análise pode ser feita agora?
- **Análise Descritiva**: Calcular estatísticas básicas (média, mediana, desvio padrão) para Age e Fare para sobreviventes e não sobreviventes. Contar a frequência de Sex, Pclass e Embarked.

- **Análise de Sobrevivência por Categoria**: Investigar a taxa de sobrevivência com base em:

  - **Gênero (Sex)**: Qual sexo teve maior taxa de sobrevivência?
  - **Classe do Passageiro (Pclass)**: Passageiros de qual classe tinham maior probabilidade de sobreviver?
  - **Porto de Embarque (Embarked)**: Há alguma diferença na taxa de sobrevivência entre os portos?
- **Relação Idade/Tarifa e Sobrevivência**: Explorar se a idade e a tarifa (Fare) pagas influenciaram a sobrevivência. Por exemplo, crianças e idosos tiveram mais chances? Tarifa mais alta significava maior chance?

###Essas análises ajudarão a obter insights sobre os fatores que podem ter influenciado a sobrevivência no Titanic.